releases/release-4.0.16.md

```diff
@@ -0,0 +1,98 @@
+---
+title: TiDB 4.0.16 Release Notes
+---
+
+# TiDB 4.0.16 Release Notes
+
+Release Date: December 10, 2021
+
+TiDB version: 4.0.16
+
+## Bug Fixes
+
++ TiDB
+
+    - Check step overflow when converting a range to points for estimation [#30017](https://github.com/pingcap/tidb/pull/30017)
+    - Fix wrong result for control function with enum type. [#30011](https://github.com/pingcap/tidb/pull/30011)
+    - expression: fix different results for greatest when vectorized is off. [#29916](https://github.com/pingcap/tidb/pull/29916)
+    - Fix index join panic on prefix index on some cases [#29217](https://github.com/pingcap/tidb/pull/29217)
+    - Planner: fix the issue that planner may cache invalid plans for joins in some cases [#28444](https://github.com/pingcap/tidb/pull/28444)
+    - fix a bug that can not insert null into a not null column in the empty SQL mode [#27832](https://github.com/pingcap/tidb/pull/27832)
+
++ TiFlash
+
+    - Fix the issue that comparison between Decimal may cause overflow and report `Can't compare`. [#3365](https://github.com/pingcap/tics/pull/3365)
+    - Fix the issue that TiFlash fails to start up under platform without library `nsl` [#3209](https://github.com/pingcap/tics/pull/3209)
+
++ PD
+
+    - Fix panic issue after TiKV node scales in [#4378](https://github.com/tikv/pd/pull/4378)
+    - Fix the issue that PD may not elect leader as soon as leader step down [#4219](https://github.com/tikv/pd/pull/4219)
+    - `evict-leader-scheduler` supports schedule the regions with unhealthy peers. [#4133](https://github.com/tikv/pd/pull/4133)
+
+## __unsorted
+
++ TiDB
+
+    - Fix wrong result type for greatest/least [#29911](https://github.com/pingcap/tidb/pull/29911)
+    - This is a fix for grant global level case in v4.0 and v5.0 [#29899](https://github.com/pingcap/tidb/pull/29899)
+    - Prevent conflicted optimistic transactions from locking each other. [#29775](https://github.com/pingcap/tidb/pull/29775)
+    - Fix panic for caseWhen function with enum type [#29508](https://github.com/pingcap/tidb/pull/29508)
+    - Fix wrong result of microsecond function in vectorized [#29384](https://github.com/pingcap/tidb/pull/29384)
+    - Fix incomplete log information about auto analyze. [#29227](https://github.com/pingcap/tidb/pull/29227)
+    - Fix wrong result of hour function in vectorized expression [#28870](https://github.com/pingcap/tidb/pull/28870)
+    - Fix an issue that NO_ZERO_IN_DATE does not work on the default values. [#26902](https://github.com/pingcap/tidb/pull/26902)
+    - Fix copt-cache metrics, it will display the number of  hits/miss/evict on Grafana. [#26342](https://github.com/pingcap/tidb/pull/26342)
+    - Fix the issue that concurrently truncating the same partition hangs DDL. [#26237](https://github.com/pingcap/tidb/pull/26237)
```

Comment:
```suggestion
## Bug Fixes

+ TiDB

    - Fix a query panic caused by overflow in statistics module when calculating selectivity and converting a range to points for cost estimation [#30017](https://github.com/pingcap/tidb/pull/30017)
    - Fix wrong result for control function (`if`, `case when`, etc) with enum type. [#30011](https://github.com/pingcap/tidb/pull/30011)
    - Fix different results for function `greatest` when vectorized is off. [#29916](https://github.com/pingcap/tidb/pull/29916)
    - Fix index join panic on prefix index on some cases [#29217](https://github.com/pingcap/tidb/pull/29217)
    - Fix a bug that planner may cache invalid plans for joins in some cases [#28444](https://github.com/pingcap/tidb/pull/28444)
    - Fix a bug that TiDB can not insert null into a not null column when `sql_mode` is empty [#27832](https://github.com/pingcap/tidb/pull/27832)
    - Fix wrong result type for function `greatest`/`least` [#29911](https://github.com/pingcap/tidb/pull/29911)
    - Fix privilege checking for `grant` and `revoke` operation on global level privileges [#29899](https://github.com/pingcap/tidb/pull/29899)
    - Fix panic for `case when` function with enum type [#29508](https://github.com/pingcap/tidb/pull/29508)
    - Fix wrong result of `microsecond` function in vectorized expression [#29384](https://github.com/pingcap/tidb/pull/29384)
    - Fix wrong result of `hour` function in vectorized expression [#28870](https://github.com/pingcap/tidb/pull/28870)
    - Prevent conflicted optimistic transactions from locking each other. [#29775](https://github.com/pingcap/tidb/pull/29775)
    - Fix incomplete log information about auto analyze. [#29227](https://github.com/pingcap/tidb/pull/29227)
    - Fix an issue that `NO_ZERO_IN_DATE` does not work on the default values. [#26902](https://github.com/pingcap/tidb/pull/26902)
    - Fix copt-cache metrics, it will display the number of hits/miss/evict on Grafana now. [#26342](https://github.com/pingcap/tidb/pull/26342)
    - Fix the issue that concurrently truncating the same partition hangs DDL. [#26237](https://github.com/pingcap/tidb/pull/26237)

+ TiFlash

    - Fix the issue that comparison between Decimal may cause overflow and report `Can't compare`. [#3365](https://github.com/pingcap/tics/pull/3365)
    - Fix the issue that TiFlash fails to start up under platform without library `nsl` [#3209](https://github.com/pingcap/tics/pull/3209)

+ PD

    - Fix panic issue after TiKV node scales in [#4378](https://github.com/tikv/pd/pull/4378)
    - Fix the issue that PD may not elect leader as soon as leader step down [#4219](https://github.com/tikv/pd/pull/4219)
    - `evict-leader-scheduler` supports schedule the regions with unhealthy peers. [#4133](https://github.com/tikv/pd/pull/4133)

## __unsorted
```

Reply:
TiDB `__unsorted` part are moved to Bug fixes and tiny change of wording

Reply:
> 

Could you please synchronize these changes to the Chinese release notes (https://github.com/pingcap/docs-cn/pull/7836) as well? Thanks.

Reply:
Done https://github.com/pingcap/docs-cn/pull/7836#discussion_r763702484